crates/walrus-core/src/encoding/quilt_encoding.rs

```diff
@@ -1749,6 +1749,30 @@ impl<'a> QuiltDecoderV1<'a> {
         }
     }
 
+    /// Creates a new QuiltDecoderV1 with owned slivers.
+    pub fn new_with_owned(slivers: impl IntoIterator<Item = SliverData<Secondary>>) -> Self {
+        let slivers: HashMap<SliverIndex, Cow<'a, SliverData<Secondary>>> = slivers
+            .into_iter()
+            .map(|s| (s.index, Cow::Owned(s)))
+            .collect();
+        let column_size = slivers.values().next().map(|s| s.symbols.data().len());
+        Self {
+            slivers,
+            quilt_index: None,
+            column_size,
+        }
+    }
+
+    /// Adds owned slivers to the decoder.
+    pub fn add_owned_slivers(&mut self, slivers: impl IntoIterator<Item = SliverData<Secondary>>) {
+        for sliver in slivers {
+            self.column_size
+                .get_or_insert_with(|| sliver.symbols.data().len());
+            let index = sliver.index;
+            self.slivers.insert(index, Cow::Owned(sliver));
+        }
+    }
```

Comment:
Question/suggestion: Can we document that we require all slivers to be of equal size and that this is not checked? Alternatively, we could check all sliver sizes.